homeassistant/components/harmony/remote.py

```diff
@@ -405,94 +319,21 @@ async def async_send_command(self, command, **kwargs):
             return
 
-        device_id = None
-        if device.isdigit():
-            _LOGGER.debug("%s: Device %s is numeric", self.name, device)
-            if self._client.get_device_name(int(device)):
-                device_id = device
-
-        if device_id is None:
-            _LOGGER.debug(
-                "%s: Find device ID %s based on device name", self.name, device
-            )
-            device_id = self._client.get_device_id(str(device).strip())
-
-        if device_id is None:
-            _LOGGER.error("%s: Device %s is invalid", self.name, device)
-            return
-
         num_repeats = kwargs[ATTR_NUM_REPEATS]
         delay_secs = kwargs.get(ATTR_DELAY_SECS, self.delay_secs)
         hold_secs = kwargs[ATTR_HOLD_SECS]
-        _LOGGER.debug(
-            "Sending commands to device %s holding for %s seconds "
-            "with a delay of %s seconds",
-            device,
-            hold_secs,
-            delay_secs,
+        await self._data.async_send_command(
+            command, device, num_repeats, delay_secs, hold_secs
         )
 
-        # Creating list of commands to send.
-        snd_cmnd_list = []
-        for _ in range(num_repeats):
-            for single_command in command:
-                send_command = SendCommandDevice(
-                    device=device_id, command=single_command, delay=hold_secs
-                )
-                snd_cmnd_list.append(send_command)
-                if delay_secs > 0:
-                    snd_cmnd_list.append(float(delay_secs))
-
-        _LOGGER.debug("%s: Sending commands", self.name)
-        try:
-            result_list = await self._client.send_commands(snd_cmnd_list)
-        except aioexc.TimeOut:
-            _LOGGER.error("%s: Sending commands timed-out", self.name)
-            return
-
-        for result in result_list:
-            _LOGGER.error(
-                "Sending command %s to device %s failed with code %s: %s",
-                result.command.command,
-                result.command.device,
-                result.code,
-                result.msg,
-            )
-
     async def change_channel(self, channel):
         """Change the channel using Harmony remote."""
-        _LOGGER.debug("%s: Changing channel to %s", self.name, channel)
-        try:
-            await self._client.change_channel(channel)
-        except aioexc.TimeOut:
-            _LOGGER.error("%s: Changing channel to %s timed-out", self.name, channel)
+        self._data.change_channel(channel)
 
     async def sync(self):
         """Sync the Harmony device with the web service."""
-        _LOGGER.debug("%s: Syncing hub with Harmony cloud", self.name)
-        try:
-            await self._client.sync()
-        except aioexc.TimeOut:
-            _LOGGER.error("%s: Syncing hub with Harmony cloud timed-out", self.name)
-        else:
-            await self.hass.async_add_executor_job(self.write_config_file)
-
-    def write_config_file(self):
```

Comment:
Grepping through the code, I could find no current reference to code that actually _reads_ this outputted config file.

I looked through the history of this component and it appears this was previously part of the old python library this component used for the harmony api, prior to moving to `aioharmony`. I think in the process of moving libraries, the previous authors started having this entity write this out since the new library didn't contain that functionality.

Apologies if there is a common pattern I was missing here, but I could find no reason to continue writing this out. Removing this simplified writing more tests around the state handling of the entities in this integration..

Reply:
The conf file is used by humans to figure out which buttons to put in their automations